js/grocy.js

```diff
@@ -212,6 +214,24 @@ class GrocyClass
 		this.IdleTime += 1;
 	}
 
+	Use(componentName)
```

Comment:
Would a similar functionality be possible with `import` ing the components. Because that could also run some code on import (although this is discouraged) but then the methods itself wouldn't be on a global but instead would be accessed from the import. And what would you think about that?

Reply:
In principle, yes, that would be possible. However, this means that importing is done statically, can't be guarded by ifs and that breaks with the userfields component which – depending on the defined fields – dynamically decides if it needs to initialize the datetimepicker component.

Some components, like the product card, already expect to be only once in the DOM, which is violated by example opening the product card from the stock overview page, opening the stock entries from the card, and opening another product card. What jquery does in that instance is pretty much undefined, but [it seems to pick up the first instance only](https://jsfiddle.net/1d2tLwzb/). I can't tell if it's problematic, but scoping solves the issue.

However the scope needs to be passed dynamically, and this can only be done when opening the modal. Scoping isn't implemented in the components yet, but I'll do the same trick I'm currently throwing at the viewjs/ folder by defining `$scope = $(container-selector).find` for scoped components / views and `$scope = $` otherwise. Unfortunately, this requires me to manually check each jquery usage in every view file, because the object returned from `$(selector)` and `$(selector).find(foobar)` aren't exactly the same type.